File: sBuilding/Trunk/Client/sBuilding/sTimeKeeping/Interface/ITimeKeepingUserTimeConfig.cs
using sTimeKeeping.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace sTimeKeeping.Interface
{
    /// <summary>
    /// interface ITimeKeepingUserTimeConfig
    /// </summary>
    public interface ITimeKeepingUserTimeConfig
    {
        /// <summary>
        /// insert UserTimeConfig
        /// </summary>
        /// <param name="session"></param>
        /// <param name="userTimeConfig"></param>
        /// <returns></returns>
        int insert(string session, UserTimeConfig userTimeConfig);
        /// <summary>
        /// update UserTimeConfig
        /// </summary>
        /// <param name="session"></param>
        /// <param name="userTimeConfig"></param>
        /// <returns></returns>
        int update(string session, UserTimeConfig userTimeConfig);
        /// <summary>
        /// delete UserTimeConfig
        /// </summary>
        /// <param name="session"></param>
        /// <param name="shList"></param>
        /// <returns></returns>
        int delete(string session, List<long> shList);
        /// <summary>
        /// get UserTimeConfig By Id
        /// </summary>
        /// <param name="session"></param>
        /// <param name="id"></param>
        /// <returns></returns>
        UserTimeConfig getUserTimeConfigById(string session, long id);
        /// <summary>
        /// get List UserTimeConfig By MemberId
        /// </summary>
        /// <param name="session"></param>
        /// <param name="orgId"></param>
        /// <param name="memberId"></param>
        /// <returns></returns>
        List<List<UserTimeConfig>> getListUserTimeConfigByMemberId(string session, long orgId, List<long> memberId);
    }
}

File: sBuilding/Trunk/Server/Java/timekeepingservice/src/com/swt/timekeeping/ITimekeepingUserTimeConfig.java
package com.swt.timekeeping;

import java.util.List;

import com.swt.timekeeping.domain.UserTimeConfig;

/** interface ITimekeepingUserTimeConfig 
 * @author Trang-PC
 */
public interface ITimekeepingUserTimeConfig {
	/**
	 * insert
	 * @param userTimeConfig
	 * @return
	 */
	public UserTimeConfig insert(UserTimeConfig userTimeConfig);
	/**
	 * update
	 * @param userTimeConfig
	 * @return
	 */
	public UserTimeConfig update(UserTimeConfig userTimeConfig);
	/**
	 * delete
	 * @param shList
	 * @return
	 */
	public int delete(List<Long> shList);
	/**
	 * get UserTimeConfig By Id
	 * @param id
	 * @return
	 */
	public UserTimeConfig getUserTimeConfigById(long id);
	/**
	 * get List UserTimeConfig By MemberId
	 * @param orgId
	 * @param memberId
	 * @return
	 */
	public List<UserTimeConfig> getListUserTimeConfigByMemberId(long orgId, long memberId);
	/**
	 * delete List UserTimeConfig by OrgId
	 * @param orgId
	 */
	public void deleteListUserTimeConfigOrgId(long orgId);
	
}

